### PR TITLE
fix: token.icon possibly null, update validators and types

### PR DESCRIPTION
**Pull request by infiniteflower**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29065?quickstart=1)

This PR fixes an issue where `token.icon` returned by the Bridge API status endpoint could be `null`, which caused validation to fail. It updates the types and the validation to match what the Bridge API returns.

## **Related issues**

Related to https://github.com/MetaMask/metamask-extension/pull/28899

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by micaelae — DISMISSED** [review]

File: app/scripts/controllers/bridge-status/validators.test.ts [hunk 1]
```diff
@@ -173,56 +210,61 @@ describe('validators', () => {
   describe('bridgeStatusValidator', () => {
     // @ts-expect-error - it.each is a function
     it.each([
+      // {
```
Comment (jclancy93) [inline app/scripts/controllers/bridge-status/validators.test.ts:213]: Should we remove this instead of commenting out?

**Review by jclancy93 — DISMISSED** [review]

**Review by ejwessel — DISMISSED** [review]

File: app/scripts/controllers/bridge-status/validators.test.ts (hunk 1 above)
Comment (infiniteflower) [reply app/scripts/controllers/bridge-status/validators.test.ts:213]: Whoops good catch, I commented out during development, will put back in

**Comment by MainS2020** [comment]
> 
> 
> ## **Description**
> 
> 
> 
> [Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29065?quickstart=1)
> 
> This PR fixes an issue where `token.icon` returned by the Bridge API status endpoint could be `null`, which caused validation to fail. It updates the types and the validation to match what the Bridge API returns.
> 
> ## **Related issues**
> 
> Related to https://github.com/MetaMask/metamask-extension/pull/28899
> 
> ## **Manual testing steps**
> 
> 1. Go to this page...
> 2.
> 3.
> 
> ## **Screenshots/Recordings**
> 
> 
> 
> ### **Before**
> 
> 
> 
> ### **After**
> 
> 
> 
> ## **Pre-merge author checklist**
> 
> - [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
> - [ ] I've completed the PR template to the best of my ability
> - [ ] I’ve included tests if applicable
> - [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
> - [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
> 
> ## **Pre-merge reviewer checklist**
> 
> - [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
> - [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.
>

File: app/scripts/controllers/bridge-status/validators.test.ts (hunk 1 above)
Comment (infiniteflower) [reply app/scripts/controllers/bridge-status/validators.test.ts]: Done in [46e7e8d](https://github.com/MetaMask/metamask-extension/pull/29065/commits/46e7e8df781082210d5d59ae084449dfec854f7b)

**Review by MainS2020 — COMMENTED** [review]
Ok

**Review by micaelae — APPROVED** [review]

**Review by jclancy93 — APPROVED** [review]

**Pull request merged**